src/ray/core_worker/core_worker.cc

```diff
@@ -3391,7 +3406,11 @@ void CoreWorker::AddSpilledObjectLocationOwner(
     // object. Add the dynamically created object to our ref counter so that we
     // know that it exists.
     RAY_CHECK(!generator_id->IsNil());
-    reference_counter_->AddDynamicReturn(object_id, *generator_id);
+    if (task_manager_->ObjectRefStreamExists(*generator_id)) {
```

Comment:
NOTE: This can have a subtle edge cases (reference leak) if 

- generator is deleted
- Users get the generator ref (it is returned when there's a system error) and saves it somewhere 
- and when these RPCs are received "after" all ^ happens.

I think this edge case is extremely rare. We can fix it better by changing the local_object_manager, but it requires a lot of code change. I am open to this option if you don't like this approach.

Reply:
Can you convert this to a comment with an example explaining the edge case?